langchain/document_loaders/sitemap.py

```diff
@@ -110,7 +115,7 @@ def load(self) -> List[Document]:
         return [
             Document(
                 page_content=self.parsing_function(results[i]),
-                metadata={**{"source": els[i]["loc"]}, **els[i]},
+                metadata={**{"source": els[i]["loc"]}, **self.meta_function(els[i], results[i])},
```

Comment:
can this just be `metadata=self.meta_function(els[i], results[i])`?

Reply:
here i was not sure if i should keep the source as field that always exists or if i should move the loc to source copy to the meta function.
i can move this into the default meta function and add a comment above to remember to extract source yourself if you overwrite it